proposals/lambdas-01-intro.md

```diff
@@ -0,0 +1,26 @@
+Yield blocks use a syntax of `->` to attach the yield block to a call site. Async yield blocks will use a similar syntax of `...->`, creating what is effectively a lambda as known in other languages.
```

Comment:
I don't think changing a symbol associated with behaviors would be enough because an objectified lambda that is not passed to a behavior should also have the syntax that indicates "called later" to the reader.

I am open to exploring other arrow options. For the current choice what I like about it is that it is very visually distinct and evocative for the reader, even if admittedly it is a bit more laborious to type for the writer.

I don't think I'm a fan where very small visual differences in the arrow type (`->` vs `=>`) are significant and commonplace because such small elements are very easy to visually blend together with one's eyes when reading code.

Reply:
that's fair, I'm not sure it's really more evocative than `->` is for lambdas, which is the norm in most other programming languages for general lambdas